solution-4.py

```diff
@@ -0,0 +1,24 @@
+import csv
+data = [
+    ["sid", "sname", "salary"],
+    [1, "jay", 3800],
+    [2, "om", 4000],  
+    [3, "sahu", 4500],
+    [4, "hiren", 4100],
+    [5, "sai", 4200]   
+]
+with open('salary.csv', 'w', newline='') as f:
+    writer = csv.writer(f)
+    writer.writerows(data)
+s_names = 0
+total_records = 0
+with open('salary.csv', 'r') as f:
+    reader = csv.reader(f)
+    header = next(reader)  

```

Comment:
can you explain why you used header=next(reader)?

Reply:
The line "header = next(reader)" reads the header row of the CSV file, which holds column names. This helps understand the data's structure when working with subsequent rows.